extensions/amp-animation/0.1/css-expr-ast.js

```diff
@@ -27,6 +90,65 @@ const DATA_URL_RE = /^data\:/i;
  */
 export class CssNode {
   constructor() {}
+
+  /**
+   * A return a string CSS representation.
+   * @return {string}
+   * @abstract
+   */
+  css() {}
+
+  /**
+   * Resolves the value of all variable components. Only performs any work if
```

Comment:
Thanks. Added more docs to explain. But basically, this is a public final method it provides a optimization hook to immediately resolve the node with itself if it's a constant (`isConst() == true`). And `calc()` is used to actually do calculations. Most of nodes are constants so they follow the shorter path with no additional memory allocations.